### DISCUSSION TITLE
Import notes from Quillpad

### DISCUSSION THREAD

**Discussion by dremelts** [TechbeeAT/jtxBoard#1084] [General]
I want to migrate my Quillpad notes into jtx notes.  I like Quillpad but it absolutely does not sync so my notes are ONLY on my phone and not accessible from my linux laptop.  Do I have any options to move my stuff to jtx?

**Comment by rogercreagh** [comment]
I believe Quillpad has an experimental sync with Nextcloud Notes facility - so in essence it is the same problem as wanting to sync with Nextclloud Notes.

Nextcloud notes has an API so in principle it is possible to write an app to run on a server to move material from Notes into CalDAV - and possible the other way allowing some form of sync to be cobbled together.

I can't find on my phone where quillpad stores its notes (but then I haven't used it since jtxB became available) - it does have an export function which creates a json file in downloads, so in principle an app could parse that and convert it to CalDav format, but it might be quite a lot of work for a one-time use. 

My solution was simply to copy and paste the text of quillpad notes directly in jtxB - but I probably had less than a twenty that I wanted to keep - if you've a load of them then perhaps it would be worth writing a script to parse the json file - but with all the overhead of an android app it would be a labour of love.

If I needed to do it (or was being paid commercial rates!) I would write it as a web server application that imported the json and wrote it directly to the CalDAV server. But then I already have the CalDAV end of that working so it would be simpler to do when I had the need. 

For myself though I'm more inclined to do the Nextcloud Notes conversion as I thnk that would be more useful for the wholle Journal/Notes ecosystem.

**Reply by dremelts** [reply]
Yes, I am aware of the "experimental" NC sync option in QuillPad but it does not work.  I also took a look at the json file that QuillPad exports and I could probably write a script to parse it into an ics file and import it into jtxB.  But, I will probably just do as you did and cut/paste because I don't really have enough notes to make it worth the effort to write a script.  I was just hoping that there was a solution already out there.

**Reply by dremelts** [reply]
I found a way to fairly easily transfer notes from QuillPad to jtxB.  If you long-press a note in QP it brings up a context menu and one of the options is Share and one of the apps you can share with is jtxB.  The only problem is that it only shares the body of the note and it does not include the title.  When you share this with jtxB it assumes that the first line of info that is being shared is the Summary and the rest of the text is the Description.  So, I had to do a little bit of fix-up on my notes but it was quite painless.  It will even Categorize your notes if you have hashtags embedded in the note.

**Reply by patrickunterwegs** [reply]
Hi @dremelts , sorry for the late reply, but anyway it seems like you already found a good way to import your notes. An alternative would also be to import notes from files which you could do in the collections overview by clicking on the context menu of a collection. But in this way you'd need to find a way to export every note from the other application to a txt-file.

Yes, the app is doing some parsing when you use this quick-add functionality (which is also used when you share a text, image or file with the app). Maybe I should re-consider if it was a good idea to interpret the first paragraph as summary and the rest as description, I see that in many cases this might not be the desired behavior...

**Reply by dremelts** [reply]
@patrickunterwegs "But in this way you'd need to find a way to export every note from the other application to a txt-file." Which is precisely the problem I am having with QuillPad - it is not syncing with NC Notes so the txt or md files are not up-to-date.  Anyway, the sharing method I used worked great for the small number of notes I needed to migrate.
